### ISSUE TITLE
`no-void` should have an option to allow void arrow functions or void async IIFEs

### ISSUE THREAD

**Issue by slikts** [eslint/eslint#12688]
The `void` operator is commonly regarded as a legacy language feature and is disabled in popular style guides (standard/standard#1464, airbnb/javascript#2145), but it does have [legit and current use cases][yes-void]. Mainly, it can be used to make non-value-returning arrow functions explicit:
```js
const log = x => void console.log(x);
```
The `no-void` rule should have an option to allow `void` in arrow functions like this.

To a lesser degree, it could be useful to also allow it for async IIFEs.

[yes-void]: https://github.com/standard/standard/issues/1464

**Comment by kaicataldo** [comment]
Can you please fill out the issue template requested? You can find it [here](https://raw.githubusercontent.com/eslint/eslint/master/.github/ISSUE_TEMPLATE/RULE_CHANGE.md). Thanks.

**Comment by G-Rath** [comment]
I have created a new issue based off this one that follows the issue template: #13020
